File: practices/24-agent-sdk-intro/src/python/basic_agent.py
"""
Practice 24: Basic Agent — Minimal Python agent using Claude Agent SDK

This demonstrates the simplest possible agent: a single query() call
that sends a prompt and receives a response.
"""

import sys
from claude_agent_sdk import Agent


def main():
    """Create a minimal agent and send a single query."""

    # Create an agent instance
    # The SDK automatically picks up ANTHROPIC_API_KEY from the environment
    agent = Agent()

    try:
        # Send a simple query
        # The agent will use available tools (Read, Bash, etc.) to fulfill the request
        result = agent.query(
            prompt="List the files in the current directory and briefly describe what each one does.",
            system_prompt={"preset": "claude_code"},
        )

        # Print the agent's response
        print("Agent response:")
        print(result.text)

        # Print usage statistics
        print(f"\n--- Usage ---")
        print(f"Input tokens:  {result.usage.input_tokens}")
        print(f"Output tokens: {result.usage.output_tokens}")

    except Exception as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()

File: practices/24-agent-sdk-intro/src/typescript/basic-agent.ts
/**
 * Practice 24: Basic Agent — Minimal TypeScript agent using Claude Agent SDK
 *
 * This demonstrates the simplest possible agent: a single query() call
 * that sends a prompt and receives a response.
 */

import { Agent } from "@anthropic-ai/claude-agent-sdk";

async function main(): Promise<void> {
  // Create an agent instance
  // The SDK automatically picks up ANTHROPIC_API_KEY from the environment
  const agent = new Agent();

  try {
    // Send a simple query
    const result = await agent.query({
      prompt:
        "List the files in the current directory and briefly describe what each one does.",
      systemPrompt: { preset: "claude_code" },
    });

    // Print the agent's response
    console.log("Agent response:");
    console.log(result.text);

    // Print usage statistics
    console.log("\n--- Usage ---");
    console.log(`Input tokens:  ${result.usage.inputTokens}`);
    console.log(`Output tokens: ${result.usage.outputTokens}`);
  } catch (error) {
    console.error("Error:", error);
    process.exit(1);
  }
}

main();
